lib/block_view_access_groups.go

```diff
@@ -0,0 +1,1026 @@
+		return errors.Wrapf(RuleErrorAccessKeyNameTooShort, "ValidateGroupPublicKeyAndName: "+
+			"Too few characters in key name: min = %v, provided = %v",
+			MinMessagingKeyNameCharacters, len(keyName))
+	}
+	if len(keyName) > MaxMessagingKeyNameCharacters {
+		return errors.Wrapf(RuleErrorAccessKeyNameTooLong, "ValidateGroupPublicKeyAndName: "+
+			"Too many characters in key name: max = %v; provided = %v",
+			MaxMessagingKeyNameCharacters, len(keyName))
+	}
+	return nil
+}
+
+// ValidateKeyAndNameWithUtxo validates public key and key name, which are used in DeSo V3 Messages protocol.
+// The function first checks that the key and name are valid and then fetches an entry from UtxoView or DB
+// to check if the key has been previously saved. This is particularly useful for connecting V3 messages.
+func (bav *UtxoView) ValidateKeyAndNameWithUtxo(ownerPublicKey, messagingPublicKey, keyName []byte, blockHeight uint32) error {
+	// First validate the public key and name with ValidateGroupPublicKeyAndName
+	err := ValidateGroupPublicKeyAndName(messagingPublicKey, keyName)
+	if err != nil {
+		return errors.Wrapf(err, "ValidateKeyAndNameWithUtxo: Failed validating "+
+			"messagingPublicKey and keyName")
+	}
+
+	// Fetch the messaging key entry from UtxoView.
+	messagingGroupKey := NewAccessGroupKey(NewPublicKey(ownerPublicKey), keyName)
+	// To validate a messaging group key, we try to fetch the simplified group entry from the membership index.
+	messagingGroupEntry := bav.GetMessagingGroupForMessagingGroupKeyExistence(
+		messagingGroupKey, blockHeight, false)
+	if messagingGroupEntry == nil || messagingGroupEntry.isDeleted {
+		return fmt.Errorf("ValidateKeyAndNameWithUtxo: non-existent messaging key entry "+
+			"for ownerPublicKey: %s", PkToString(ownerPublicKey, bav.Params))
+	}
+
+	// Compare the UtxoEntry with the provided key for more validation.
+	if !reflect.DeepEqual(messagingGroupEntry.AccessPublicKey[:], messagingPublicKey) {
+		return fmt.Errorf("ValidateKeyAndNameWithUtxo: keys don't match for "+
+			"ownerPublicKey: %s", PkToString(ownerPublicKey, bav.Params))
+	}
+
+	if !EqualGroupKeyName(messagingGroupEntry.AccessGroupKeyName, NewGroupKeyName(keyName)) {
+		return fmt.Errorf("ValidateKeyAndNameWithUtxo: key name don't match for "+
+			"ownerPublicKey: %s", PkToString(ownerPublicKey, bav.Params))
+	}
+	return nil
+}
+
+// getGroupMemberAttributeEntry returns the group member attribute entry for the given group member.
+func (bav *UtxoView) getGroupMemberAttributeEntry(groupOwnerPublicKey *PublicKey, groupKeyName *GroupKeyName,
+	memberPublicKey *PublicKey, attributeType AccessGroupMemberAttributeType) (*AttributeEntry, error) {
+	// Create enumeration key.
+	enumerationKey := NewGroupEnumerationKey(groupOwnerPublicKey, groupKeyName[:], memberPublicKey)
+	// Check if enumerationKey exists in GroupMemberAttributes mapping.
+	attributeEntry, exists := bav.GroupMemberAttributes[*enumerationKey][attributeType]
```

Comment:
Then the AttributeType for that enumerationKey would also not exist, making the above logic skip over the utxoView and checking in the DB. Tested the nested map logic in go playground :)

Something to note: Even though the AttributeType won't exist for a non-existent enumerationKey, when asked for a value for the non-existent key in the map, it would return a valid zero-value. So we definitely need to check the "exists" bool instead of value content to do this correctly.